app/models/miq_request/purging.rb

```diff
@@ -0,0 +1,32 @@
+class MiqRequest
+  module Purging
+    extend ActiveSupport::Concern
+    include PurgingMixin
+
+    module ClassMethods
+      def purge_date
+        ::Settings.miq_request.history.keep_miq_requests.to_i_with_method.seconds.ago.utc
+      end
+
+      def purge_window_size
+        ::Settings.miq_request.history.purge_window_size
+      end
+
+      def purge_scope(older_than)
+        MiqRequest.where(arel_table[:created_on].lt(older_than))
+      end
+
+      # this is MiqRequest#destroy hook (defined by relationships on MiqRequest)
+      def purge_associated_records(ids)
+        # sorry for the destroy. this needs to be recursive:
+        MiqRequestTask.where(:miq_request_id => ids).destroy_all
+        MiqApproval.where(:miq_request_id => ids).delete_all
+        RequestLog.where(:resource_id => ids).delete_all
```

Comment:
if we're using purge_method destroy, then either these should not be marked dependent => destroy or they should be marked dependent destroy, and this line removed.

Reply:
Or perhaps remove this entire purge_associated_records method and just let them get destroyed naturally

Reply:
Yeah, I had removed this method entirely and put it back.  I feel like instantiating hundreds of RequestLog objects per request could be problematic.  I've seen 300-800 request logs per requests in various databases.

I was debating dropping the top two (and let destroy handle it) and leaving the last one since the others seem to be 1:1 in terms of requests:tasks and requests:approvals.

Reply:
I supposed I can change has_many request_logs to `dependent => :delete_all`

Reply:
Now that request_logs is dependent => delete_all I don't see what this is buying us now